test/run-integration-tests.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash -e
+
+if ! test -d ../build ; then
+        echo "Expected build directory in ../build, but couldn't find it." >&2
+        exit 1
+fi
```

Comment:
It seems strange that we don't already have script like that. I wonder what the Ubuntu autopkgtests are doing. Aha, they are using a manual loop: https://anonscm.debian.org/cgit/pkg-systemd/systemd.git/tree/debian/tests/upstream. So they should be reworked to used this at some point.

Please use `test/Makefile.guess` instead of hardcoding `build`.

Reply:
Makefile.guess doesn't really work. It gets confused by mkosi.builddir which also is a build dir, but definitely nothing the test scripts should care about. Quite frankly I think we should drop any such guessing work, and just stick to "build" as builddir. I see no value in using different possible values. I didn't want to add code for dropping the builddir to this PR though...

Reply:
For example the meson build macros in Fedora use a different directory. If `Makefile.guess` doesn't work, we should fix it. But I agree that this doesn't have to happen now.

Reply:
oh, they do? inside of the source tree? or somewhere else? do you know why they use a different one?

Reply:
```console
$ rpmbuild --eval '%meson_build'
    /usr/bin/ninja -v -j4 -C x86_64-redhat-linux-gnu
$ rpmbuild --eval '%meson_install'
    DESTDIR=/home/zbyszek/rpmbuild/BUILDROOT/%{NAME}-%{VERSION}-%{RELEASE}.x86_64 /usr/bin/ninja install -v -j4 -C x86_64-redhat-linux-gnu
```

I assume that it's to allow different architectures to build from the same directory without conflicts.

Reply:
ah, interesting. good to know.  i wonder if they specify the build dir in some env var however... if so we should just use that if it is set instead of guessing...

Reply:
It's available as rpm macros (`%{_vpath_srcdir}`, `%{_vpath_builddir}`). But I'd rather fix autodetection. I guess it's just a question of skipping the mkosi dirs. I'll give it a go.